Documentation updates for file backend integration

JoeARO opened the pull request:
This contains changes to the documentation giving a configuration
guide specifically how to use a custom backend for logging to a file.

Due to the inheritance through config files I have also changed the
top level `config.exs` to contain the non test related time_api variable,
however I an empty `dev.exs` is required, otherwise `iex -S mix` throws an error.

JoeARO committed:
Documentation updates for file backend integration

This contains changes to the documentation giving a configuration
guide specifically how to use a custom backend for logging to a file.

Due to the inheritance through config files I have also changed the
top level `config.exs` to contain the non test related time_api variable,
however I an empty `dev.exs` is required, otherwise `iex -S mix` throws an error.

samfrench commented on README.md:
```diff
@@ -51,3 +52,33 @@ It can also be nicely used in conjuction with libraries such as [HTTPoison](http
+## Configuration
+
+The default configuration for this will simply log to the console, if you would like to configure it you can simply edit your `config.exs` file.
+It is worth noting that you must keep the format as `format: "$message\n"` with whichever logging backend you choose to use otherwise you will get duplication of information.
```
Why `format: message` and why does the information get duplicated?

samfrench replied:
How do we guarantee the messages are in a hash/json format? If a single string is used, does this get logged?

JoeARO replied:
Sam the [ReadMe](https://github.com/bbc/mozart-fetcher/pull/10) already has an example of a string being used instead of a map to show we can use that.
The format has to be format: "$message\n" otherwise you will end up with the JSON plus the datetime/level duplicated but outside of that.

```elixir
iex(1)> EventLogger.log(:error, "foo")

14:50:39.992 [error] {"message":"foo","level":"error","datetime":"2019-03-13T14:50:39.987369Z"}
:ok
```

samfrench replied:
The readme in the PR  linked in the previous comment seems to be for the logging. Is there a particular part that I should look at for the string example? The examples in the readme look like component requests but I may have missed your point.

Not sure why the datetime would be duplicated without a "\n". Did you find out why it was doing this in your investigations?

samfrench commented on config/config.exs:
```diff
@@ -29,8 +28,10 @@ use Mix.Config
-
 config :event_logger, :environment, Mix.env()
 
+config :event_logger, time_api: EventLogger.Time.DateTime
```
Is this format UTC?

JoeARO replied:
Yes sam if you look at the function here you can see that it just returns the results of DateTime.utc_now

https://github.com/JoeARO/event_logger/blob/master/lib/event_logger/time/date_time.ex#L6

This was implemented so the time could be mocked for testing

samfrench replied:
I assumed it was. Just checking in case we needed to set GMT or specify a UK timezone somewhere.

JoeARO committed:
Rewording of some sentences in ReadMe

Some of the sentences didnt quite make sense so have
reworded/restructured so they read easier.
